consensus/README.md

```diff
@@ -140,6 +140,222 @@ graph TB
   J --> note2
 ```
 
+## Synchronization between the consensus processes
+
+The consensus module currently depends on the `PaceMaker`, `StateSync`, `LeaderElection`, `Networking`.
+It has a bootstrapping state where it:
+  * Initializes connections to the network through a bootstrap node
+  * Gets the latest block then switches to one of the mutually exclusive modes: `sync` or `consensus`
+  * Keeps an updated current height (the greatest block height seen on the network)
+
+```mermaid
+sequenceDiagram
+participant Consensus
+participant StateSync
+Note over Node: This embeds Networking, FSM
+participant Node
+participant Network
+Node->>Node:Start
+Node->>Network:Join
+loop
+  Node->>Network:RequestCurrentHeight
+  Network-->>Node: 
+  Node->>Node:SwitchMode
+end
+alt SyncMode
+  par
+    Node->>Consensus:Pause
+  and
+    Node->>StateSync:Start
+    Note over Node,StateSync: Details omitted for another diagram
+    StateSync-->>Node:SyncDone
+  end
+else ConsensusMode
+  par
+    Node->>StateSync:Pause
+  and
+    Node->>Consensus:Start
+    Consensus-->>Node:OutOfSync
+  end
+end
```

Comment:
Great diagram!

I just edited newlines, whitespaces, spacing and section activations to improve readability. Can you do the same below?

```suggestion
sequenceDiagram
    participant Consensus
    participant StateSync
    Note over Node: This embeds Networking, FSM
    participant Node
    participant Network

    Node->>Node: Start
    Node->>Network: Join
    loop
        Node->>+Network: RequestCurrentHeight
        Network-->>-Node: 
        Node->>Node: SwitchMode
    end
    
    alt SyncMode
        par
            Node->>Consensus: Pause
        and
            Node->>+StateSync: Start
            Note over Node,StateSync: Details omitted for another diagram
            StateSync-->>-Node: SyncDone
        end
    else ConsensusMode
        par
            Node->>StateSync:Pause
        and
            Node->>+Consensus:Start
            Consensus-->>-Node:OutOfSync
        end
    end
```